app/Providers/AppServiceProvider.php

```diff
@@ -195,4 +202,24 @@ public function registerFeatures()
             return new Features($app[ConfigRepository::class]);
         });
     }
+
+    private function registerBusses(): void
+    {
+        $this->app->singleton(CommandBus::class, function ($app) {
+            $commandBus = new CommandBus($app);
+
+            $commandBus->register(ExampleCommand::class, ExampleCommandHandler::class);
```

Comment:
I am considering we have an automated way to register actions to the bus, maybe specified in a config file.

```
// config/cqrs.php
<?php
return [
   'queries' => [
        \Ushahidi\App\Bus\Query\Example\ExampleQuery::class => \Ushahidi\App\Bus\Query\Example\ExampleQueryHandler::class,
 ],
 
  'commands' => [
        \Ushahidi\App\Bus\Command\Example\ExampleCommand::class => \Ushahidi\App\Bus\Command\Example\ExampleCommandHandler::class,
 ],
];
```

Then in the `registerBusses` function in service provider we have this to register queries (vice versa for commands)

```
foreach (config('cqrs.queries') as $action => $handler) {
        $queryBus->register($action, $handler);
 }
```

Reply:
I don't think it's any more automated than what's in there really. But adds on complexity as there more files to keep track of.

Reply:
In the scenario, we have about 10 - 20 commands and queries to register, one would have to register in the AppServiceProvider making it a large file to work with, this is what I look at avoiding. Also, we only have one file `cqrs.php` in the config folder, where any one can go and register an action to the bus.

About complexity, I really don't think so ... this is a pattern widely adopted in the laravel ecosystem i.e the `app.php` uses such approach in registering providers.

Reply:
I agree with @webong about this point,  this file might be a quite large after adding many commands
& queries